test/HubPool.RefundExecution.ts

```diff
@@ -0,0 +1,81 @@
+import { expect } from "chai";
+import { Contract } from "ethers";
+import { ethers } from "hardhat";
+
+import { SignerWithAddress, toBNWei, seedWallet } from "./utils";
+import * as consts from "./constants";
+import { hubPoolFixture, enableTokensForLiquidityProvision } from "./HubPool.Fixture";
+import { buildPoolRebalanceTree, buildPoolRebalanceLeafs } from "./MerkleLib.utils";
+
+let hubPool: Contract, mockAdapter: Contract, weth: Contract, dai: Contract, mockSpoke: Contract, timer: Contract;
+let owner: SignerWithAddress, dataWorker: SignerWithAddress, liquidityProvider: SignerWithAddress;
+let l2Weth: string, l2Dai: string;
+
+describe("HubPool Relayer Refund Execution", function () {
+  beforeEach(async function () {
+    [owner, dataWorker, liquidityProvider] = await ethers.getSigners();
+    ({ weth, dai, hubPool, mockAdapter, mockSpoke, timer, l2Weth, l2Dai } = await hubPoolFixture());
+    await seedWallet(dataWorker, [dai], weth, consts.bondAmount.add(consts.finalFee).mul(2));
+    await seedWallet(liquidityProvider, [dai], weth, consts.amountToLp.mul(10));
+
+    await enableTokensForLiquidityProvision(owner, hubPool, [weth, dai]);
+    await weth.connect(liquidityProvider).approve(hubPool.address, consts.amountToLp);
+    await hubPool.connect(liquidityProvider).addLiquidity(weth.address, consts.amountToLp);
+    await dai.connect(liquidityProvider).approve(hubPool.address, consts.amountToLp.mul(10)); // LP with 10000 DAI.
+    await hubPool.connect(liquidityProvider).addLiquidity(dai.address, consts.amountToLp.mul(10));
+  });
+
+  it("Execute relayer refund correctly produces the refund bundle call and sends cross-chain repayment actions", async function () {
```

Comment:
this test covers the main logic added in this PR. please read through it to see how the leafs are created, tree constructed, refund initiated and finally executed.